File: FRONT-END/legacy-tracking-solution/TrackingUtils/Constants/Patterns.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TrackingUtils.Constants
{
    public class Patterns
    {
        public const string terminal_command = @"^\[[A-Za-z0-9]{2}\*[0-9]+\*[0-9A-Fa-f]{4}\*[A-Za-z0-9]{2,}(,.+)?\]$";
        public const string terminal_command_list = @"^(\[[A-Za-z0-9]{2}\*[0-9]+\*[0-9A-Fa-f]{4}\*[A-Za-z0-9]{2,}(,.+)?\])+$";
        public const string terminal_command_find = @"\[[A-Z0-9a-z]{2}\*[0-9]+\*[0-9A-Fa-f]{4}\*[A-Za-z0-9]{2,}(,.+)?\]";
        public const string phonenumber = @"^((\+|00)\d{1,3}|0)\d{10}$";
        public const string timeperiod = @"^\d{2}:\d{2}-\d{2}:\d{2}$";
        public const string timeperiods = @"^(\d{2}:\d{2}-\d{2}:\d{2})(,\d{2}:\d{2}-\d{2}:\d{2})*$";
        public const string reminder = @"^\d{2}:\d{2}-\d-\d(-\d{7})?$";
        public const string number = @"^\d+$";
    }
}

File: FRONT-END/TrackApp/src/constants/patterns.js
export const PHONE_NUMBER = /^\+?\d{10,14}$/;
export const DEVICE_PASSWORD = /^[a-zA-Z0-9]{6,}$/;
export const NUMBER = /^\d+$/;
export const CONTACT_NAME = /^[a-zA-Z0-9\s]+$/;
export const IP = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$/;
export const HOST_NAME = /^([a-z0-9A-Z]\.)*[a-z0-9-]+\.([a-z0-9]{2,24})+(\.co\.([a-z0-9]{2,24})|\.([a-z0-9]{2,24}))*$/;